src/index.rs

```diff
@@ -133,25 +194,25 @@ impl ResourceIndex {
                     } else {
                         log::debug!("Removing {:?} from index", meta.id);
                         self.ids.remove(&meta.id);
-                        deleted.insert(meta.id);
+                        deleted.insert(path, meta);
                     }
                 } else {
                     log::warn!("Path {} was not known", path.display());
                 }
             });
 
         let added: HashMap<CanonicalPathBuf, ResourceMeta> =
-            scan_metadata(updated_paths)
+            scan_metadata(&updated_paths)
                 .into_iter()
                 .chain({
                     log::info!("The same for new paths");
-                    scan_metadata(created_paths).into_iter()
+                    scan_metadata(&created_paths).into_iter()
                 })
                 .filter(|(_, meta)| !self.ids.contains(&meta.id))
                 .collect();
 
         for (path, meta) in added.iter() {
-            if deleted.contains(&meta.id) {
+            if deleted.contains_key(path) {
```

Comment:
No, this doesn't look correct. Log message bellow: "Resource _id_ was moved to _path_".

If `path` is contained in both `added` and `deleted` then we have a mistake in their calculations and must emit something like `AssertionError`.

Resource is indeed moved only if the a resource with `meta.id` was deleted at `path1` and was added at `path2`.